Packs/CyberArkPAS/Integrations/CyberArkISP/CyberArkISP.py

```diff
@@ -0,0 +1,679 @@
+
+    demisto.debug(f"[Date Helper] Final parsed date: {parsed_datetime.isoformat()}")
+    return parsed_datetime
+
+
+def generate_telemetry_header() -> str:
+    """Generate the base64-encoded telemetry header value.
+
+    Returns:
+        Base64-encoded telemetry string
+    """
+    telemetry_data = (
+        f"in={INTEGRATION_NAME}&"
+        f"it={Config.INTEGRATION_TYPE}&"
+        f"iv={Config.INTEGRATION_VERSION}&"
+        f"vn={Config.VENDOR_NAME}&"
+        f"VV={Config.VENDOR_VERSION}"
+    )
+    encoded = base64.b64encode(telemetry_data.encode()).decode()
+    demisto.debug(f"[Telemetry] Generated header: {telemetry_data}")
+    return encoded
+
+
+def parse_integration_params(params: dict[str, Any]) -> dict[str, Any]:
+    """Parse and validate integration configuration parameters."""
+    demisto.debug("[Config] Starting parameter validation")
+
+    base_url = (params.get("url", "")).strip().rstrip("/")
+    if not base_url:
+        raise DemistoException("Server URL is required. Please provide the Audit API base URL.")
+
+    identity_url = (params.get("identity_url", "")).strip().rstrip("/")
+    if not identity_url:
+        raise DemistoException("Identity URL is required. Please provide the CyberArk Identity FQDN.")
+
+    web_app_id = params.get("web_app_id", "").strip()
+    if not web_app_id:
+        raise DemistoException("OAuth2 Web App ID is required.")
+
+    client_id = params.get("client_id", "").strip()
+    if not client_id:
+        raise DemistoException("Client ID is required.")
+
+    credentials = params.get("credentials", {})
+    client_secret = credentials.get("password", "").strip()
+    if not client_secret:
+        raise DemistoException("Client Secret is required.")
+
+    api_key = params.get("api_key", "").strip()
+    if not api_key:
+        raise DemistoException("API Key is required.")
+
+    proxy = argToBoolean(params.get("proxy", False))
+    verify_certificate = not argToBoolean(params.get("insecure", False))
+
+    # Construct token URL
+    token_url = f"{identity_url}/OAuth2/Token/{web_app_id}"
+
+    demisto.debug(f"[Config] Base URL: {base_url} | Token URL: {token_url}")
+
+    return {
+        "base_url": base_url,
+        "token_url": token_url,
+        "verify": verify_certificate,
+        "proxy": proxy,
+        "client_id": client_id,
+        "client_secret": client_secret,
+        "api_key": api_key,
+    }
+
+
+def add_time_to_events(events: list[dict[str, Any]]) -> None:
+    """Add _time field to events for XSIAM ingestion.
+
+    Maps the event's 'timestamp' field to '_time' for proper XSIAM indexing.
+    The value is copied as-is without any parsing or transformation.
+    """
+    for event in events:
+        event_timestamp = event.get("timestamp")
+        if event_timestamp:
+            event["_time"] = event_timestamp
+        else:
+            demisto.debug(f"[Event Time] WARNING: Event missing 'timestamp' field: {event.get('uuid', 'unknown')}")
```

Comment:
Events without timestamps won't have _time field, which could cause XSIAM ingestion issues.
Lets consider setting to current time.